packages/core/src/strategies/basic-strategy.ts

```diff
@@ -6,43 +7,79 @@ import {
   executeOnchainInteractionRequest,
   executeStaticCallRequest,
   getStaticCallExecutionStateResultValue,
-} from "./execution-strategy-helpers";
-import { ExecutionResultType } from "./types/execution-result";
+} from "../internal/execution/execution-strategy-helpers";
+import { JsonRpcClient } from "../internal/execution/jsonrpc-client";
+import { ExecutionResultType } from "../internal/execution/types/execution-result";
 import {
   CallExecutionState,
   DeploymentExecutionState,
   SendDataExecutionState,
   StaticCallExecutionState,
-} from "./types/execution-state";
+} from "../internal/execution/types/execution-state";
 import {
   CallStrategyGenerator,
   DeploymentStrategyGenerator,
   ExecutionStrategy,
-  LoadArtifactFunction,
   OnchainInteractionResponseType,
   SendDataStrategyGenerator,
   StaticCallStrategyGenerator,
-} from "./types/execution-strategy";
-import { NetworkInteractionType } from "./types/network-interaction";
+} from "../internal/execution/types/execution-strategy";
+import { NetworkInteractionType } from "../internal/execution/types/network-interaction";
+import { assertIgnitionInvariant } from "../internal/utils/assertions";
 
 /**
- * The most basic execution strategy, which sends a single transaction
- * for each deployment, call, and send data, and a single static call
- * per static call execution.
+ * The basic execution strategy, which sends a single transaction
+ * for each contract deployment, call, and send data, and a single static call
+ * for each static call execution.
+ *
+ * @beta
  */
-export class BasicExecutionStrategy implements ExecutionStrategy {
-  public readonly name: string = "basic";
+export class BasicStrategy {
+  public readonly name = "basic" as const;
+  public readonly config: Record<PropertyKey, never>;
```

Comment:
We initially had `{}`, but that implies a far broader type than the empty object, hence the odd record construct above.